### PR TITLE
Enable automated code coverage reports via coveralls.io

### LINKED ISSUE #22: Enable automated code coverage reports
**What is the expected feature or enhancement?**
I like the use of https://coveralls.io/ in https://github.com/Qiskit/qiskit-terra and other qiskit packages for reviewing code coverage.

**Goal**
Let's discuss and decide as part of this issue if we want something similar (e.g. ability to review code coverage over time, to see comments in PRs with coverage details [[example]](https://github.com/Qiskit/qiskit-terra/pull/7288#issuecomment-972798040)) in the provider repositories.

See also
- https://coveralls.io/github/Qiskit
- https://github.com/Qiskit/qiskit-terra/issues/771

### PR DESCRIPTION
**Pull request by daka1510**
### Summary
Enable Coveralls integration. See below links for details.
- [Coveralls Github Action](https://github.com/marketplace/coveralls)
- [Coveralls Github Action Source](https://github.com/coverallsapp/github-action)

Inspired by other open source Qiskit libraries that already integrate with Coveralls (https://coveralls.io/github/Qiskit)

### Details and comments
Partially solves #22. We should discuss adding a code coverage badge to the README once this PR is integrated to fully solve it.

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -24,7 +24,7 @@ qiskit-aer
 websockets>=8
 scikit-quant;platform_system != 'Windows'
 black==21.11b1
-coverage>=6.2
+coverage>=6.3
```
Comment (daka1510) [inline requirements-dev.txt:27]: Support for `lcov` reports was added in 6.3

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -92,11 +92,13 @@ jobs:
...
-      - name: Upload unit test coverage report
-        uses: actions/upload-artifact@v2
+      - name: Report coverage to coveralls.io
+        uses: coverallsapp/github-action@1.1.3
         with:
-          name: Unit test coverage report - python${{ matrix.python-version }}-${{ matrix.os }}
-          path: htmlcov
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          flag-name: unit-tests_python${{ matrix.python-version }}-${{ matrix.os }}
+          parallel: true
```
Comment (daka1510) [inline .github/workflows/ci.yml:100]: Given that our coverage differs based on environment, test type and potentially other factors I like the ability to report coverage metrics for all jobs in a single run.

Shows up in Coveralls like below
image

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -133,8 +135,20 @@ jobs:
...
-      - name: Upload integration test coverage report
-        uses: actions/upload-artifact@v2
+      - name: Report coverage to coveralls.io
+        uses: coverallsapp/github-action@1.1.3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          flag-name: integration-tests_${{ matrix.environment }}
+          parallel: true
+          path-to-lcov: coverage.lcov
+  tests-finished:
```
Comment (rathishcholarajan) [inline .github/workflows/ci.yml:145]: Could you please add a name for this section so it looks good in [this](https://github.com/daka1510/qiskit-ibm-runtime/actions/runs/1844766846) view?

**Review by rathishcholarajan — APPROVED** [review]
LGTM! Left a minor suggestion, can be merged after addressing that...

File: .github/workflows/ci.yml (hunk 3 above)
Comment (daka1510) [reply .github/workflows/ci.yml:145]: See https://github.com/Qiskit/qiskit-ibm-runtime/pull/144/commits/c0b416243f43a47be2c2009a042820d49a1831a7

**Pull request merged**